src/SprykerEco/Zed/Payone/Business/PayoneBusinessFactory.php

```diff
@@ -175,143 +471,101 @@ protected function createKeyHashGenerator()
+    public function createTransactionStatusRequest(PayoneTransactionStatusUpdateTransfer $transactionStatusUpdateTransfer): TransactionStatusRequest
     {
         return new TransactionStatusRequest($transactionStatusUpdateTransfer->toArray());
     }
 
     /**
-     * @return array
-     */
-    protected function getAvailablePaymentMethods()
-    {
-        $storeConfig = $this->getProvidedDependency(PayoneDependencyProvider::STORE_CONFIG);
-
-        return [
-            PayoneApiConstants::PAYMENT_METHOD_CREDITCARD_PSEUDO => $this->createCreditCardPseudo($storeConfig),
-            PayoneApiConstants::PAYMENT_METHOD_INVOICE => $this->createInvoice($storeConfig),
-            PayoneApiConstants::PAYMENT_METHOD_SECURITY_INVOICE => $this->createSecurityInvoice($storeConfig),
-            PayoneApiConstants::PAYMENT_METHOD_ONLINE_BANK_TRANSFER => $this->createOnlineBankTransfer($storeConfig),
-            PayoneApiConstants::PAYMENT_METHOD_E_WALLET => $this->createEWallet($storeConfig),
-            PayoneApiConstants::PAYMENT_METHOD_PREPAYMENT => $this->createPrepayment($storeConfig),
-            PayoneApiConstants::PAYMENT_METHOD_DIRECT_DEBIT => $this->createDirectDebit($storeConfig),
-            PayoneApiConstants::PAYMENT_METHOD_PAYPAL_EXPRESS_CHECKOUT => $this->createGenericPayment($storeConfig),
-            PayoneApiConstants::PAYMENT_METHOD_CASH_ON_DELIVERY => $this->createCashOnDelivery($storeConfig, $this->getGlossaryFacade()),
-        ];
-    }
-
-    /**
-     * @return \Generated\Shared\Transfer\PayoneStandardParameterTransfer
-     */
-    protected function getStandardParameter()
-    {
-        if ($this->standardParameter === null) {
-            $this->standardParameter = $this->getConfig()->getRequestStandardParameter();
-        }
-
-        return $this->standardParameter;
-    }
-
-    /**
-     * @return \SprykerEco\Zed\Payone\Business\Key\HashProvider
+     * @return \SprykerEco\Zed\Payone\Business\Payment\RequestSender\PayoneBaseAuthorizeSenderInterface
      */
-    protected function createHashProvider()
+    public function createPayoneBaseAuthorizeSender(): PayoneBaseAuthorizeSenderInterface
     {
-        $hashProvider = new HashProvider();
-
-        return $hashProvider;
+        return new PayoneBaseAuthorizeSender(
+            $this->createExecutionAdapter(),
+            $this->getQueryContainer(),
+            $this->createPaymentMapperReader(),
+            $this->getStandardParameter(),
+            $this->createStandartParameterMapper(),
+        );
     }
 
     /**
      * @param \Spryker\Shared\Kernel\Store $storeConfig
      *
      * @return \SprykerEco\Zed\Payone\Business\Payment\MethodMapper\CreditCardPseudo
      */
-    protected function createCreditCardPseudo($storeConfig)
+    public function createCreditCardPseudo(Store $storeConfig): PaymentMethodMapperInterface
```

Comment:
* Fix the return type hint in the document block.
* Check other methods.
* Fix the code sniffer.

Reply:
++